### DISCUSSION TITLE
[QUESTION] Is it possible to change crack width (w_lim) using IOM?

### DISCUSSION THREAD

**Discussion by kniazmajowiec** [idea-statica/ideastatica-public#287] [Help Q&A]
Is it possible to change crack width (w_lim) using IOM? 

Going through the github I found the SetupValueEc2 corresponding to the table 7.1N (iom/IdeaRS.OpenModel/Concrete/ConcreteSetupEc2.cs):
![image](https://user-images.githubusercontent.com/52321719/227516582-bc7f9ce5-a1b4-4ed6-9a4f-d436e7dc7b52.png)

However, I do not know how to assign table value as SetupValue

I will be grateful for a solution.

**Comment by nathanlukeIDEA** [comment]
Hi @kniazmajowiec,

Apologies for the slow response. Please put future issues in this repo.

Does the below code help to set the value you mention?

```csharp
//Concrete setup
OpenModel openModel = new OpenModel();

var setup = new ConcreteSetupEc2();
setup.Annex = NationalAnnexCode.Dutch;
openModel.ConcreteSetup = setup;
```

Thanks 
nathan

**Comment by kniazmajowiec** [comment]
Dear Nathan,

Thank you for the answer. 
It is a solution; however, I would prefer to be able to use the National Annex of my choice but still limit the crack width. 

I was able to change some values (for example NA_7_3_4_K3, see the code snippet below), but those were doubles (SetupValueDouble)

```
//Concrete setup
var setup = new ConcreteSetupEc2();
setup.Annex = NationalAnnexCode.French;

var na734 = new SetupValueDouble();
double cover = Convert.ToDouble(memberDataDict["cover"], CultureInfo.InvariantCulture);
na734.Value = 3.4 * Math.Pow((25 / cover), 2.0 / 3);
na734.Id = (int)SetupValueEc2.NA_7_3_4_K3;

var setupList = new List<SetupValue>();
setupList.Add(na734);
```
      
I thought, there is maybe a solution with class NadStrategyConcrete as it takes List of NadSetupValue?
